File: 06.5_salary_increase_by_category_with_match/ocaml/bin/main.ml
let () = print_endline "\n--- PROGRAMA DE AUMENTO DE SALARIO POR CATEGORIAS ---\n"

let category = print_endline "Ingresa la categoria a la que perteneces (A, B, C):"; read_line ()

let increment_percentage = match category with
| "a" | "A" -> 0.15
| "b" | "B" -> 0.1
| "c" | "C" -> 0.07
| _ -> prerr_endline (Printf.sprintf "La categoria \"%s\" ingresada no existe" category); exit 1

let salary = print_endline "Ingresa tu salario actual:"; float_of_string (read_line ())

let increment = salary *. increment_percentage
let new_salary = salary +. increment

let () = print_endline (Printf.sprintf "\nTu aumento será de: $ %f" increment);
print_endline (Printf.sprintf "Tu nuevo salario será de: $ %f" new_salary)

File: 06.5_salary_increase_by_category_with_match/rust/src/main.rs
use std::{io, process};
use std::io::Write;

fn main() {
    let mut input = String::new();

    println!("\n--- PROGRAMA DE AUMENTO DE SALARIO POR CATEGORIAS ---\n");

    print!("Ingresa la categoria a la que perteneces (A, B, C) > ");
    io::stdout().flush().unwrap();
    io::stdin().read_line(&mut input).unwrap();
    let category = input.trim().to_string();

    let increment_percentage = match category.as_str() {
        "a" | "A" => 0.15_f32,
        "b" | "B" => 0.1,
        "c" | "C" => 0.07,
        _ => {
            eprintln!("La categoria \"{category}\" ingresada no existe");
            process::exit(1);
        }
    };

    print!("Ingresa tu salario actual > ");
    io::stdout().flush().unwrap();
    input.clear();
    io::stdin().read_line(&mut input).unwrap();
    let salary = input.trim().parse::<f32>().expect("El salario debe ser un número");

    let increment = salary * increment_percentage;
    let new_salary = salary + increment;

    println!("\nTu aumento será de: $ {increment}");
    println!("Tu nuevo salario será de: $ {new_salary}");
}
